test/test-sortedarray.c

```diff
@@ -0,0 +1,241 @@
+/*
+
+Copyright (c) 2016, Stefan Cloudt
+
+Permission to use, copy, modify, and/or distribute this software
+for any purpose with or without fee is hereby granted, provided
+that the above copyright notice and this permission notice appear
+in all copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <time.h>
+
+#include "alloc-testing.h"
+#include "framework.h"
+
+#include "sortedarray.h"
+
+#define TEST_SIZE 100
+
+int compare_float(void* v1, void* v2) 
```

Comment:
I'm not sure why you structured your tests around float arrays. The library already has functions for testing equality/comparison of ints. Floats are rather problematic because they cannot be reliably compared for equality. It would be better if you just restructured your tests to work with floats.